paimon-core/src/main/java/org/apache/paimon/jdbc/JdbcUtils.java

```diff
@@ -0,0 +1,646 @@
+                execute(connections, JdbcUtils.updatePropertiesStatement(properties.size()), args);
+        if (updatedRecords == properties.size()) {
+            return true;
+        }
+        throw new IllegalStateException(
+                String.format(
+                        "Failed to update: %d of %d succeeded", updatedRecords, properties.size()));
+    }
+
+    public static boolean deleteProperties(
+            JdbcClientPool connections,
+            String catalogName,
+            String databaseName,
+            Set<String> properties) {
+        String[] args =
+                Stream.concat(Stream.of(catalogName, databaseName), properties.stream())
+                        .toArray(String[]::new);
+
+        return execute(connections, JdbcUtils.deletePropertiesStatement(properties), args) > 0;
+    }
+
+    private static String insertPropertiesStatement(int size) {
+        StringBuilder sqlStatement = new StringBuilder(JdbcUtils.INSERT_DATABASE_PROPERTIES_SQL);
+        for (int i = 0; i < size; i++) {
+            if (i != 0) {
+                sqlStatement.append(", ");
+            }
+            sqlStatement.append(JdbcUtils.INSERT_PROPERTIES_VALUES_BASE);
+        }
+        return sqlStatement.toString();
+    }
+
+    private static String deletePropertiesStatement(Set<String> properties) {
+        StringBuilder sqlStatement = new StringBuilder(JdbcUtils.DELETE_DATABASE_PROPERTIES_SQL);
+        String values =
+                String.join(",", Collections.nCopies(properties.size(), String.valueOf('?')));
+        sqlStatement.append("(").append(values).append(")");
+
+        return sqlStatement.toString();
+    }
+
+    private static String updatePropertiesStatement(int size) {
+        StringBuilder sqlStatement =
+                new StringBuilder(
+                        "UPDATE "
+                                + DATABASE_PROPERTIES_TABLE_NAME
+                                + " SET "
+                                + DATABASE_PROPERTY_VALUE
+                                + " = CASE");
+        for (int i = 0; i < size; i += 1) {
+            sqlStatement.append(" WHEN " + DATABASE_PROPERTY_KEY + " = ? THEN ?");
+        }
+        sqlStatement.append(
+                " END WHERE "
+                        + CATALOG_NAME
+                        + " = ? AND "
+                        + DATABASE_NAME
+                        + " = ? AND "
+                        + DATABASE_PROPERTY_KEY
+                        + " IN ");
+
+        String values = String.join(",", Collections.nCopies(size, String.valueOf('?')));
+        sqlStatement.append("(").append(values).append(")");
+        return sqlStatement.toString();
+    }
+
+    public static boolean acquire(JdbcClientPool connections, String lockName, long timeout)
+            throws SQLException, InterruptedException {
+        // Check and clear expire lock
+        int affectedRows = tryClearExpireLock(connections, lockName, timeout);
```

Comment:
I think `timeout` is a property that should be set when acquiring the lock, not when the lock expires. Because different clients may configure different timeouts.